### ISSUE TITLE
Tuya devices are failing to trigger during automations or updates status if they do trigger 

### ISSUE THREAD

**Issue by BacchusIX** [tuya/tuya-home-assistant#743]
Since updating to home assistant 2021.12.2, all Tuya devices are failing to properly trigger during automations. If they do happen to trigger, or trigger via the app, the state does not change properly  which means they will often times fail the next trigger.

**Comment by Abhome** [comment]
Same here.

**Comment by fdkssdks** [comment]
Same here

**Comment by b3ckford** [comment]
Same. I found disabling and then re-enabling the integration solves the issue for short while, then it goes back to not reporting the state again.

**Comment by bmorgan99** [comment]
same. 2012.12.4

**Comment by fideco1981** [comment]
I have the same problem. Switches do trigger, but state change does not update. A reload of the integration works for a short while.

**Comment by jheld** [comment]
Confirm that the work around does mitigate the issue at least temporarily for me, too.

**Comment by tabnul** [comment]
Same here. No updates from devices get propagated to HA. Only after reloading the integration it works for a short while . In my case tuya heaters.

Basically this makes the integration unusable for me.

**Comment by jonastep** [comment]
Same issue here with light. Status of devices won´t update after reloading the integration.

**Comment by raflaisaahole** [comment]
same

**Comment by tabnul** [comment]
Seems like major bug

**Comment by Peterpc58** [comment]
same problem

**Comment by tabnul** [comment]
https://github.com/tuya/tuya-iot-python-sdk/pull/51

https://github.com/home-assistant/core/issues/62408#issuecomment-999648277

**Comment by tabnul** [comment]
solved by installing homeassistant 12.6

**Comment by psychowood** [comment]
> solved by installing homeassistant 12.6 

Same here, I'll monitor but 12.6 seems to have fixed the issue

**Comment by BacchusIX** [comment]
Yeah I never got the manual patch to work but 2021.12.6 seems to work.

**Comment by BacchusIX** [comment]
If no one else responds that they are having issues even after installing 12.6, I'll close the issue in a few days.

**Comment by fdkssdks** [comment]
Solved for me after updating HASS

**Comment by maijo366** [comment]
The same issue. I updated to HA 2021.12.8. Always the same. When i trigger some iot - Switch. The Switch turn on, but the status is of and in a few seconds the button goes back.

**Comment by tabnul** [comment]
I have comparable issues but it doesnt seem to be specifically related to tuya.  All buttons seem to stop working after a few pushes. It takes a reatart of HA app on phone to get them working again.

**Comment by Anarkistic** [comment]
Also having this issue, device states simply not reliably updating themselves, lights in particular in my case. The issue did get fixed in a recent core update that updated the SDK to 0.6.6 but then the last couple of days (since the core got updated again) it's broken again. 

Currently on core 2021.12.8 and OS 7.1

**Comment by imtiajmeah** [comment]
I'm still having the same issue.. on the latest update Home Assistant 2021.12.8. temp workaround is to reload integration but after a while, it has to be reloaded again to get the correct status. Definitely not fixed yet.

Home Assistant 2021.12.8
Host Operating System	Home Assistant OS 7.1
Update Channel	stable

**Comment by BacchusIX** [comment]
I'm still on 12.6 and not having issues that I've noticed; maybe try rolling back to that? 

That's good to know not to update yet, though.

**Comment by maijo366** [comment]
I was on 12.6. The same issue. All the ApI services on Tuya Development are correct and subscribed. Only LocalTuya ar working and send the status. 
I think many other users have the same issue. When i reload the integration the status change correct.

**Comment by wattmatters** [comment]
Tuya switch state does not work. Again.
Home Assistant 2021.12.6.

It worked for a few days after update to 2021.12.6 but has reverted to not updating switch state. Restarting HA resets the switch state, and it might work once after that but then it fails to update subsequent changes in switch state.
As with others all the requirements on the Tuya Devt platform are correct and the switch state is correctly reflected in Tuya's Smart Life app.

**Comment by clearwave1** [comment]
I went from 2021.12.4 (where it seemed fixed) to 2021.12.7 which broke it. 

I'm now on 2021.12.8 and it is still broken.

**Comment by tabnul** [comment]
I am starting to think it is a GUI issue that is left.
Totally different devices on the same dashboard behave the same (only change state 1 or 2 times, after that i have to restart the app to use the devices again). Other dashboards no issues ..

I dont have to reload.integrations.

**Comment by wattmatters** [comment]
Today my HA automation failed to turn off my Tuya switch. Will monitor to see if this was a one off or if the issue continues tomorrow.

**Comment by clearwave1** [comment]
For anyone interested, I removed this integration and replaced it with Local Tuya. The setup is a little more involved but I've had none of the issues described above and the communication is local so it is more reliable.
